plugins/src/host/plugin_handle/wasm.rs

```diff
@@ -315,10 +317,10 @@ where
             .map_err(|e| {
                 CompilationError::ModuleCouldNotCreateContract(path.clone(), c.clone(), e.into())
             })?;
-        let buf = self.read_to_vec(result_ptr)?;
+        let buf = String::from_utf8(self.read_to_vec(result_ptr)?).unwrap(); // TODO: is unwrap OK here?
```

Comment:
it might be good to encapsulate the bytes to struct pathway (not generically, just some fixed encoder types)

so that these complex changes are easier to refactor later.

Reply:
unwrap is probably not OK -- it'd be better to have an error if the data is not a UTF8 String, previously this would be handled in the json deserializer.